### PR TITLE
change blindedBlockCreatorCache keys from slot to slotAndBlockRoot.

### LINKED ISSUE #8625: [SSV] Fix cache behaviour when requesting blocks in parallel
There are scenarios where multiple operators may be connected to the same Beacon node,each of which requests the Beacon block proposal, in case they become the leader and must propose it on SSV. We have some cases in code, which cache data by slot for block proposal (with the usual assumption only one block proposal is requested per slot) so in case of two responses (one builder response and one fallback), there could be issues. On first look, the only two cases are in `ExecutionLayerBlockProductionManagerImpl` and in `FailoverValidatorApiHandler`.
 
More background on Discord and an issue occuring: https://discord.com/channels/697535391594446898/697539289042649190/1286291762167545958

### PR DESCRIPTION
**Pull request by david-ry4n**
## PR Description
Part of fixes for SSV compatibility. Switch cache key from slot to slotAndBlockRoot.

## Fixed Issue(s)

Fixes #8625 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: validator/api/src/main/java/tech/pegasys/teku/validator/api/ValidatorApiChannel.java [hunk 1]
```diff
@@ -93,95 +93,97 @@ public SafeFuture<Optional<PeerCount>> getPeerCount() {
...
         public SafeFuture<Void> registerValidators(
-            SszList<SignedValidatorRegistration> validatorRegistrations) {
+            final SszList<SignedValidatorRegistration> validatorRegistrations) {
           return SafeFuture.COMPLETE;
         }
 
         @Override
         public SafeFuture<Optional<List<ValidatorLivenessAtEpoch>>> getValidatorsLiveness(
-            List<UInt64> validatorIndices, UInt64 epoch) {
+            final List<UInt64> validatorIndices, final UInt64 epoch) {
```
Comment (StefanBratanov) [inline validator/api/src/main/java/tech/pegasys/teku/validator/api/ValidatorApiChannel.java:186]: Nice adding of final haha

File: validator/remote/src/main/java/tech/pegasys/teku/validator/remote/FailoverValidatorApiHandler.java [hunk 2]
```diff
@@ -254,12 +261,14 @@ public SafeFuture<List<SubmitDataError>> sendAggregateAndProofs(
...
-    if (blockContainer.isBlinded() && blindedBlockCreatorCache.containsKey(slot)) {
-      final ValidatorApiChannel blockCreatorApiChannel = blindedBlockCreatorCache.remove(slot);
+    final SlotAndBlockRoot slotAndBlockRoot = blockContainer.getSignedBlock().getSlotAndBlockRoot();
+    if (blockContainer.isBlinded() && blindedBlockCreatorCache.containsKey(slotAndBlockRoot)) {
+      final ValidatorApiChannel blockCreatorApiChannel =
+          blindedBlockCreatorCache.remove(slotAndBlockRoot);
       LOG.info(
-          "Block for slot {} was blinded and will only be sent to the beacon node which created it.",
-          slot);
+          "Block for slot {} and block {} was blinded and will only be sent to the beacon node which created it.",
```
Comment (StefanBratanov) [inline validator/remote/src/main/java/tech/pegasys/teku/validator/remote/FailoverValidatorApiHandler.java]: We mention block twice, I think it's cleaner to say `Block for slot {} and root {}`

**Review by StefanBratanov — APPROVED** [review]
LGTM. The failing test seems easy to fix, will let you figure it out :)

**Comment by tbenr** [comment]
I'll ask to one of you why this is needed :)

**Comment by StefanBratanov** [comment]
> I'll ask to one of you why this is needed :)

Essentially, it is a bit of protection for future enhancements for zero cost. If we implement replays, best block produced from multiple BN functionalities. Also I think it's more concise. Similar to `isLocallyCreatedBlock` in the ValidatorApiHandler.

**Comment by tbenr** [comment]
> > I'll ask to one of you why this is needed :)
> 
> Essentially, it is a bit of protection for future enhancements for zero cost. If we implement replays, best block produced from multiple BN functionalities. Also I think it's more concise. Similar to `isLocallyCreatedBlock` in the ValidatorApiHandler.

ok, sold

**Pull request merged**